pkg/controller/controller.go

```diff
@@ -327,12 +338,76 @@ func (c *Controller) DeleteMasterPool(clusterName string) error {
 }
 
 // DeleteComputePool deletes a compute node pool.
-func (c *Controller) DeleteComputePool(clusterName, name string) error {
+func (c *Controller) DeleteComputePool(clusterName string, names ...string) error {
 	pooler, impl := c.Cloud.NodePooler()
 	if !impl {
 		return ErrNotImplemented
 	}
 
-	c.Logger.Printf("deleting computepool %q of cluster %q", name, clusterName)
-	return pooler.DeleteComputePool(clusterName, name)
+	for _, name := range names {
+		c.Logger.Printf("deleting computepool %q of cluster %q", name, clusterName)
+		err := pooler.DeleteComputePool(clusterName, name)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func filterMasterPools(pools []*model.MasterPool, names []string) []*model.MasterPool {
+	pool := []*model.MasterPool{}
```

Comment:
filtered is an adjective while filter is a verb. I'd normally prefer a verb for a function